src/containers/ListingPage/SectionYoutubeVideoMaybe.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+import css from './ListingPage.module.css';
+import classNames from 'classnames';
+
+import { lazyLoadWithDimensions } from '../../util/uiHelpers.js';
+
+import { AspectRatioWrapper } from '../../components/index.js';
+import { Heading } from '../../components';
+
+const SectionYoutubeVideoMaybe = props => {
+  const { videoUrl, heading } = props;
+
+  const extractYouTubeID = url => {
+    const regExp = /(?:youtube\.com\/(?:[^\/\n\s]+\/\S+\/|(?:v|e(?:mbed)?|shorts)\/|\S*?[?&]v=)|youtu\.be\/)([a-zA-Z0-9_-]{11})/i;
+    const match = url.match(regExp);
+
+    return match ? match[1] : null;
+  };
+
+  const extractedVideoID = extractYouTubeID(videoUrl);
+
+  const RADIX = 10;
+  const BLACK_BG = '#000000';
+
+  const IFrame = props => {
+    const { dimensions, ...rest } = props;
+    return <iframe {...dimensions} {...rest} />;
+  };
+  const LazyIFrame = lazyLoadWithDimensions(IFrame);
+
+  const YoutubeEmbed = props => {
```

Comment:
Move all of these components (IFrame, LazyIFrame, YoutubeEmbed) out of the **SectionYoutubeVideoMaybe**. (Define them above it.)

If you define them inside the rendering loop of a component, they will be defined from scratch every time the wrapping component rerenders. That's an antipattern.